cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -140,20 +143,19 @@
   // injected by cdap
   @SuppressWarnings("unused")
   private Metrics workflowMetrics;
+  private ETLBatchConfig config;
 
-  private BatchPipelineSpec spec;
```

Comment:
almost all of the changes go away if you keep this variable and just set it in the configure method instead of in the constructor.